pkg/operator/ceph/cluster/controller.go

```diff
@@ -979,3 +996,72 @@ func validateExternalClusterSpec(cluster *cluster) error {
 
 	return nil
 }
+
+func validateCephVersionsBetweenLocalAndExternalClusters(localVersion, externalVersion cephver.CephVersion, otherCRPresent bool) error {
+	// Identical version, regardless if other CRs are running, it's ok!
+	if cephver.IsIdentical(localVersion, externalVersion) {
+		return nil
+	}
+
+	// external cluster version is 15.2.1 and local version is 14.2.3, and rgws / mds / nfs are running
+	// regardless if other CRs are running, the external cluster must never be higher than the
+	// it's NOT ok, you MUST update immediatly!
+	if localVersion.Major < externalVersion.Major {
+		return fmt.Errorf("external cluster ceph version is a major version higher %s than the local cluster %s", externalVersion.String(), localVersion.String())
+	}
+
+	// external cluster version is 14.2.1 and local version is 15.2.3
+	// regardless if other CRs are running, the local cluster version must never be higher than the external one
+	// You just picked the wrong image to configure your external cluster CR
+	// it's NOT ok, you MUST update immediatly!
+	if localVersion.Major > externalVersion.Major {
+		return fmt.Errorf("external cluster ceph version is a major version lower %s than the local cluster %s", externalVersion.String(), localVersion.String())
+	}
+
+	// external version is 14.2.2 and local version is 14.2.1, and rgws / mds / nfs are NOT running
+	// This assumes we just bootstrapped and it's not really acceptable
+	// So you should update the cluster CR image
+	if !otherCRPresent && localVersion.Major == externalVersion.Major {
+		if cephver.IsSuperior(externalVersion, localVersion) {
+			return fmt.Errorf("external cluster ceph version is a minor version higher %s than the local cluster %s", externalVersion.String(), localVersion.String())
```

Comment:
- What if they're only using rbd? Those other CRs won't be present and we won't have any local daemons besides the csi driver. 
- Why is it important to check this case? Seems like it should be a warning just like the a few lines lower. Then we wouldn't need to detect otherCRPresent either.

Reply:
I added a check for Block Pools so that we are 100% sure nothing is present and this is an initial bootstrap.